TombEngine/Objects/TR3/Entity/Shiva.cpp

```diff
@@ -286,19 +286,18 @@ namespace TEN::Entities::Creatures::TR3
 		{
 			if (item->Animation.ActiveState != SHIVA_STATE_DEATH)
 			{
-				SetAnimation(item, SHIVA_ANIM_DEATH);
+				SetAnimation(*item, SHIVA_ANIM_DEATH);
 				item->ItemFlags[0] = object.nmeshes - 1;
-				item->ItemFlags[2] = 1; // Redo mesh swap to stone.
+				item->ItemFlags[2] = 1; // Do mesh swap to stone.
 				item->ItemFlags[3] = 1;
 			}
 
-			int frameEnd = GetAnimData(object, SHIVA_ANIM_DEATH).frameEnd - 1;
-			if (item->Animation.FrameNumber >= frameEnd)
+			if (TestLastFrame(*item))
 			{
-				// Block frame until mesh is swapped.
-				item->Animation.FrameNumber = frameEnd;
+				// Block last frame until mesh is swapped.
+				item->Animation.FrameNumber -= 1;
 
```

Comment:
Are you sure that the condition is correct?

Reply:
Yes, this makes the Shiva work correctly.